src/table_properties.rs

```diff
@@ -0,0 +1,273 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crocksdb_ffi::{self, DBTableProperties, DBTableProperty, DBUserCollectedProperties,
+                   DBUserCollectedPropertiesIterator, DBTablePropertiesCollection,
+                   DBTablePropertiesCollectionIterator};
+use libc::size_t;
+use std::collections::HashMap;
+use std::marker::PhantomData;
+use std::slice;
+use std::str;
+
+pub struct TablePropertiesCollection {
+    pub inner: *mut DBTablePropertiesCollection,
+}
+
+impl Drop for TablePropertiesCollection {
+    fn drop(&mut self) {
+        unsafe {
+            crocksdb_ffi::crocksdb_table_properties_collection_destroy(self.inner);
+        }
+    }
+}
+
+impl TablePropertiesCollection {
+    pub fn new() -> TablePropertiesCollection {
```

Comment:
So I can create a `TablePropertiesCollection` and then call `collect` method? This will pollute the safety guarantee.